Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -524,6 +524,9 @@ theorem support_nonempty {u v : V} (p : G.Walk u v) : { w | w ∈ p.support }.No
 theorem mem_support_iff {u v w : V} (p : G.Walk u v) :
     w ∈ p.support ↔ w = u ∨ w ∈ p.support.tail := by cases p <;> simp
 
+lemma mem_support_iff' {G : SimpleGraph V} {u v w x : V} (huv : G.Adj u v) (p : G.Walk v w) :
+    x ∈ (Walk.cons huv p).support ↔ x = u ∨ x ∈ p.support := by cases p <;> simp
+
```

Comment:
It's not clear to me that this lemma needs to exist, since it's just a combination of `support_cons` and `List.mem_cons`. (More generally, if we added every lemma which is the combination of two others, the library would be way bigger for very little gain!)